src/Roslyn.Diagnostics.Analyzers/Core/ExportedPartsShouldHaveImportingConstructorCodeFixProvider.cs

```diff
@@ -0,0 +1,189 @@
+
+            var generator = SyntaxGenerator.GetGenerator(document);
+
+            var declaration = exportAttribute;
+            var declarationKind = generator.GetDeclarationKind(declaration);
+            while (declarationKind != DeclarationKind.Class)
+            {
+                declaration = generator.GetDeclaration(declaration.Parent);
+                if (declaration is null)
+                {
+                    return document;
+                }
+
+                declarationKind = generator.GetDeclarationKind(declaration);
+            }
+
+            var importingConstructor = generator.ConstructorDeclaration(
+                containingTypeName: generator.GetName(declaration),
+                parameters: Enumerable.Empty<SyntaxNode>(),
+                Accessibility.Public,
+                DeclarationModifiers.None,
+                baseConstructorArguments: null,
+                statements: Enumerable.Empty<SyntaxNode>());
+            importingConstructor = generator.AddAttributes(importingConstructor, generator.Attribute(generator.TypeExpression(importingConstructorAttributeSymbol)));
+
+            var newDeclaration = generator.AddMembers(declaration, importingConstructor);
+            return document.WithSyntaxRoot(root.ReplaceNode(declaration, newDeclaration));
+        }
+
+        private async Task<Document> MakeConstructorPublicAsync(Document document, TextSpan sourceSpan, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var importingConstructorAttribute = root.FindNode(sourceSpan, getInnermostNodeForTie: true);
+
+            var generator = SyntaxGenerator.GetGenerator(document);
+
+            var declaration = importingConstructorAttribute;
+            var declarationKind = generator.GetDeclarationKind(declaration);
+            while (declarationKind != DeclarationKind.Constructor)
+            {
+                declaration = generator.GetDeclaration(declaration.Parent);
+                if (declaration is null)
+                {
+                    return document;
+                }
+
+                declarationKind = generator.GetDeclarationKind(declaration);
+            }
+
+            var newDeclaration = generator.WithAccessibility(declaration, Accessibility.Public);
+            return document.WithSyntaxRoot(root.ReplaceNode(declaration, newDeclaration));
+        }
+
+        private async Task<Document> AddImportingConstructorAttributeAsync(Document document, TextSpan sourceSpan, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var constructor = root.FindNode(sourceSpan, getInnermostNodeForTie: true);
```

Comment:
Bail out on null `constructor`?

Reply:
Not possible unless the code fix infrastructure is broken